### ISSUE TITLE
Rule request: `not` on non-empty tuple is always `False`

### ISSUE THREAD

**Issue by tamir-bahar** [astral-sh/ruff#19739]
### Summary

I ran into the following in my code:

```python
return not (a or b or c,)
```

Due to the `,` before the closing parens, the expression was a non-empty tuple and therefore always returned `False`.

Since we have the `not` there, I doubt there will be many false-positives for that (honestly, I can't think of any).

Thanks!

**Comment by ntBre** [comment]
I could have sworn we had a rule for this, but the two I found only apply in certain contexts, which is probably what I was thinking of:
- [if-tuple (F634)](https://docs.astral.sh/ruff/rules/if-tuple/#if-tuple-f634)
- [assert-tuple (F631)](https://docs.astral.sh/ruff/rules/assert-tuple/#assert-tuple-f631)

I guess along this theme we could call the rule something like `negated-tuple`, but these all seem like special cases of a hypothetical `tuple-in-boolean-context` rule.

**Comment by MichaReiser** [comment]
or even a rule that detects values that are always truthy or falsy values in boolean context

**Comment by tamir-bahar** [comment]
Would you be open to accepting a PR that adds a rule for this specific issue?

**Comment by ntBre** [comment]
This is still labeled https://github.com/astral-sh/ruff/labels/needs-decision, so I'm not sure it's quite ready for implementation. The first decision seems to be the scope of the rule: whether it should be specific to tuples or generalized to always-truthy or always-falsy values.

I also wondered today if this could be a better fit for ty since it already has some support for reachability analysis. However, when testing this snippet in ty, it didn't seem to recognize the return value as always falsy:

```py
def foo(a, b, c):
    return not (a or b or c,)

if foo(1, 2, 3):
    ...
```

https://play.ty.dev/d2559222-a6aa-46fc-af76-3881eb8c5550

but I'm not sure how realistic it would be to detect that in ty either.

**Comment by MichaReiser** [comment]
> et in ty, it didn't seem to recognize the return value as always falsy

ty doesn't yet support return type inference.
